### PR TITLE
feat: job decorator

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Enable running jobs via decorated functions!

*Testing done:*

Unit/integ/manual

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ajberdy** [comment]
2 loc missing coverage and a few dangling todos remaining

File: src/braket/jobs/hybrid_job.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
+import functools
+import importlib.util
+import inspect
+import shutil
+import tempfile
+import warnings
+from logging import Logger, getLogger
+from pathlib import Path
+from types import ModuleType
+from typing import Any, Callable, Dict, Iterable, Optional, Union
```
Comment (speller26) [inline src/braket/jobs/hybrid_job.py:10]: We've now deprecated Python 3.8, so you no longer have to import Callable, Dict and Iterable

File: test/integ_tests/job_test_module/job_test_submodule/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+pytest
```
Comment (math411) [inline test/integ_tests/job_test_module/job_test_submodule/requirements.txt:1]: Where is this file used?

File: src/braket/jobs/hybrid_job.py (hunk 1 above)
Comment (ajberdy) [reply src/braket/jobs/hybrid_job.py:10]: Exciting, I will update this PR to use 3.9+ syntax

File: src/braket/jobs/hybrid_job.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+    try:
+        serialized = cloudpickle.dumps(wrapped_entry_point)
+    except Exception as e:
+        raise RuntimeError(
+            "Serialization failed for decorator hybrid job. For troubleshooting, "
+            "see the developer guide: #todo: link to docs with info as below"
+            # "Serialization failed for decorator hybrid job. If you are referencing "
+            # "an object from outside the function scope, either directly or through "
+            # "function parameters, try instantiating the object inside the decorated "
+            # "function instead."
```
Comment (kshitijc) [inline src/braket/jobs/hybrid_job.py:272]: Could we update this?

File: src/braket/jobs/hybrid_job.py [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+    copy_checkpoints_from_job: str | None = None,
+    checkpoint_config: CheckpointConfig | None = None,
+    role_arn: str | None = None,
+    stopping_condition: StoppingCondition | None = None,
+    output_data_config: OutputDataConfig | None = None,
+    aws_session: AwsSession | None = None,
+    tags: dict[str, str] = None,
+    logger: Logger = getLogger(__name__),
+) -> Callable:
+    """Creates a job by invoking the Braket CreateJob API.
```
Comment (kshitijc) [inline src/braket/jobs/hybrid_job.py:67]: This isn't valid if local=True. Could we update to make it unambiguous?

**Review by kshitijc — DISMISSED** [review]

File: src/braket/jobs/hybrid_job.py (hunk 4 above)
Comment (ajberdy) [reply src/braket/jobs/hybrid_job.py:67]: Good point!

**Review by kshitijc — APPROVED** [review]

**Pull request merged**